### PR TITLE
Speed up cosine affinity in kernels

### PR DESCRIPTION
**Pull request by victoris93**
I replaced `x = 1 - squareform(pdist(x, metric='cosine'))` with `x = cosine_similarity(x)` from `sklearn.metrics.pairwise`. This speeds up computation considerably when done on high-res surfaces (fsLR 32k).

### CODE REVIEW

File: brainspace/gradient/kernels.py [hunk 1]
```diff
@@ -24,7 +24,8 @@ def _build_kernel(x, kernel, gamma=None):
         return np.corrcoef(x)
 
     if kernel in {'cosine', 'normalized_angle'}:
-        x = 1 - squareform(pdist(x, metric='cosine'))
+        if kernel == 'cosine':
+            x = cosine_similarity(x)
```
Comment (OualidBenkarim) [inline brainspace/gradient/kernels.py]: normalized_angle needs the cosine similarity too, the if is not needed

```py
# if kernel == 'cosine': <-- not needed
x = cosine_similarity(x)
```

**Review by OualidBenkarim — CHANGES_REQUESTED** [review]

**Review by victoris93 — COMMENTED** [review]
Ok! Fork updated

**Review by victoris93 — COMMENTED** [review]
removed `if kernel == 'cosine'`

File: brainspace/gradient/kernels.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def _build_kernel(x, kernel, gamma=None):
         return np.corrcoef(x)
 
     if kernel in {'cosine', 'normalized_angle'}:
-        x = 1 - squareform(pdist(x, metric='cosine'))
+            x = cosine_similarity(x)
```
Comment (OualidBenkarim) [inline brainspace/gradient/kernels.py]: If you remove the extra indentation it would be good to go :)

```py
if kernel in {'cosine', 'normalized_angle'}:
    x = cosine_similarity(x) # remove extra indentation - has to be at the same level with the if below
    if kernel == 'normalized_angle':
        x = 1 - np.arccos(x, x)/np.pi
    return x
```

**Review by OualidBenkarim — CHANGES_REQUESTED** [review]

**Review by victoris93 — COMMENTED** [review]
Oops missed this one. Should be good now.

**Review by OualidBenkarim — APPROVED** [review]
Great! Thanks @victoris93
